src/Fixer/ArrayNotation/ArraySingleMultiLineFixer.php

```diff
@@ -0,0 +1,219 @@
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class ArraySingleMultiLineFixer extends AbstractFixer implements ConfigurableFixerInterface, WhitespacesAwareFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Single line arrays that are long or contain many elements must be multiline.',
+            [
+                new CodeSample(
+                    "<?php\n\$a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25];\n"
+                ),
+                new CodeSample(
+                    "<?php\n\$a = [1, 2, 3];\n",
+                    ['element_count' => 2, 'inner_length' => 10000, 'conditions' => 'any']
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before ArrayIndentationFixer, TrailingCommaInMultilineFixer.
+     * Must run after TrimArraySpacesFixer, WhitespaceAfterCommaInArrayFixer.
+     */
+    public function getPriority(): int
+    {
+        return 30;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isAnyTokenKindsFound([T_ARRAY, CT::T_ARRAY_INDEX_CURLY_BRACE_OPEN, CT::T_ARRAY_SQUARE_BRACE_OPEN]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        foreach ($this->findArrays($tokens) as $array) {
+            if ($array['is_multi_line']) {
+                continue;
+            }
+
+            $elementCountThreshold = $array['element_count'] >= $this->configuration['element_count'];
+            $innerLengthThreshold = $array['inner_length'] >= $this->configuration['inner_length'];
+
+            if ('any' === $this->configuration['conditions']) {
+                if ($elementCountThreshold || $innerLengthThreshold) {
+                    $this->expandArray($tokens, $array);
+                }
+            } elseif ($elementCountThreshold && $innerLengthThreshold) { // 'conditions' is always 'all' here
+                $this->expandArray($tokens, $array);
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        $assertIntMoreThanZero = [static function (int $value): bool { return $value > 0; }];
+
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('element_count', 'Threshold: # of elements an array must have be be written multiline.'))
+                ->setDefault(25)
```

Comment:
Can we not have these defaults, so anyone who wants to use the fixer must set the numbers by himself?